### PR TITLE
Add Hoverfly to incubator

### PR DESCRIPTION
**Pull request by tommysitu**
#### What this PR does / why we need it:

Add an incubator chart for [Hoverfly](https://github.com/SpectoLabs/hoverfly)

#### Special notes for your reviewer:
Would love to get this into the official Helm repo. Thank you for the review in advance!

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: incubator/hoverfly/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
```
Comment (mattfarina) [inline incubator/hoverfly/templates/deployment.yaml]: Can you please update to use the `apps/v1` API for deployments. The extensions API will be turned off soon. It is long deprecated.

File: incubator/hoverfly/values.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Default values for hoverfly.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+image:
+  repository: docker.io/spectolabs/hoverfly
+  tag: v1.0.0-rc.2
```
Comment (mattfarina) [inline incubator/hoverfly/values.yaml:6]: Can you please update to follow the format documented at https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md#configuration

File: incubator/hoverfly/values.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+service:
+  type: ClusterIP
+  externalAdminPort: 8888
+  adminPort: 8888
+  externalProxyPort: 8500
+  proxyPort: 8500
+healthcheckEndpoint: /api/health
+hoverflyFlags:
+resources:
```
Comment (mattfarina) [inline incubator/hoverfly/values.yaml]: Can you please comment out the resources by default.

Comment (tommysitu) [reply incubator/hoverfly/values.yaml]: Fixed

File: incubator/hoverfly/values.yaml (hunk 2 above)
Comment (tommysitu) [reply incubator/hoverfly/values.yaml:6]: Thank you for your review. I've added the `image.pullPolicy` as per the guideline. Is it good enough?

**Review by mattfarina — APPROVED** [review]

**Comment by tommysitu** [comment]
Thank you @mattfarina !

**Pull request merged**